docker/gaffer-rest/Dockerfile

```diff
@@ -16,15 +31,16 @@ WORKDIR /jars
 # If users want to provide their own utility jars they can do that within jars/lib
 COPY ./jars/ .
 
-RUN	if [ ! -f "./rest.jar" ]; then \
+RUN	if echo "$ACCUMULO_VERSION" | grep -q "^2.*$"; then LEGACY=false; else LEGACY=true; fi && \
+    if [ ! -f "./rest.jar" ] && [ "$LEGACY" = false ]; then \
 		curl -sfLo rest.jar "${GAFFER_DOWNLOAD_URL}/uk/gov/gchq/gaffer/spring-rest/${GAFFER_VERSION}/spring-rest-${GAFFER_VERSION}-exec.jar" || true; \
 	fi && \
     if [ ! -f "./rest.jar" ]; then \
 		git clone ${GAFFER_GIT_REPO} /tmp/gaffer && \
 		cd /tmp/gaffer && \
-		git checkout ${GAFFER_VERSION} && \
-		mvn clean package -q -Pquick --also-make -pl :spring-rest && \
-		cp ./rest-api/spring-rest/target/spring-rest-*-exec.jar /jars/rest.jar && \
+		git checkout ${GAFFER_VERSION} || git checkout gaffer2-${GAFFER_VERSION} && \
+		mvn clean package -Dlegacy=$LEGACY -q -Pquick --also-make -pl :spring-rest && \
+		cp ./rest-api/spring-rest/target/spring-rest-*-exec*.jar /jars/rest.jar && \
```

Comment:
Same question here - is the additional wildcard here significant / required?

Reply:
Same as above, the Accumulo 1 jars that are built with the legacy flag have `-legacy` appended, so this is to catch them.